app/src/main/java/com/example/hackernews/MainActivity.kt

```diff
@@ -139,27 +166,37 @@ class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
         })
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        binding.drawerLayout.openDrawer(Gravity.START)
+        return super.onOptionsItemSelected(item)
+    }
+
+
+
     override fun onNavigationItemSelected(item: MenuItem): Boolean {
         when(item.itemId) {
             R.id.side_top_stories -> {
                 newsAdapter.clear()
                 binding.drawerLayout.closeDrawer(Gravity.START)
-                apiCall.getStories(NewsDataType.TOP_STORIES, newsAdapter)
+                apiCall.getStories(NewsDataType.TOP_STORIES, this)
             }
             R.id.side_catch_up -> {
                 binding.drawerLayout.closeDrawer(Gravity.START)
                 newsAdapter.clear()
             }
             R.id.side_new_stories -> {
                 binding.drawerLayout.closeDrawer(Gravity.START)
-                apiCall.getStories(NewsDataType.NEW_STORIES, newsAdapter)
+                apiCall.getStories(NewsDataType.NEW_STORIES, this)
                 newsAdapter.clear()
             }
             R.id.side_feedback -> {
                 Toast.makeText(this, "Clicked on feedback", Toast.LENGTH_LONG).show()
             }
             R.id.side_saved_stories -> {
-                Toast.makeText(this, "Clicked on saved stories", Toast.LENGTH_LONG).show()
+                binding.drawerLayout.closeDrawer(Gravity.START)
+                newsAdapter.clear()
+                apiCall.stopLoadingNews()
+                AuthUser.getToken()?.let { userPostDAO.loadPosts(it, this, this.apiCall) }
```

Comment:
Since `getToken` is called and used multiple times, it would make sense to define it as an instance property and only fetch it once.